### PR TITLE
Loki: Remove legacy form classes from LokiQuerySettings

### PR DESCRIPTION
**Pull request by shabeebk**
With this commit, legacy form components are remvoed from loki query settings page. Form is now using InlineField instead.

**What is this feature?**

Removing the legacy form from the core components listed in https://github.com/grafana/grafana/issues/76107

**Why do we need this feature?**

As per the ticket https://github.com/grafana/grafana/issues/76107, these are meant to be replaced with its modern replacement.

**Who is this feature for?**

Dev

**Which issue(s) does this PR fix?**:

Relates #76107

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by matyax** [comment]
Hi @shabeebk ! Thank you for submitting this PR. Would you mind adding screenshots of the before and after UI to make sure we don't have any visual regression? That'd be much appreciated.

**Comment by shabeebk** [comment]
Sure. Will add the screenshot soon.

**Comment by shabeebk** [comment]
**Screenshots**

**Before**
Screenshot 2023-10-11 at 12 59 18

**After**
Screenshot 2023-10-11 at 13 01 02

**Comment by shabeebk** [comment]
**Screenshots**

**Before**
Screenshot 2023-10-11 at 12 59 18

**After**
Screenshot 2023-10-11 at 13 10 40

**Comment by shabeebk** [comment]
@matyax Just added the before/after images after the latest commit https://github.com/grafana/grafana/pull/76331/commits/337ae7d198e287f065ccb4d92c95b80d1b5bb834

**Comment by tskarhed** [comment]
@matyax We're moving away from inline forms, at least for the configuration pages (https://github.com/grafana/grafana/issues/66479). So I think I'd rather see the use of `Field` here, but it is your call in the end. It will change the look, and if it introduces inconsistencies that you'd rather deal with later, I completely understand that.

**Comment by matyax** [comment]
Thanks @shabeebk 

@tskarhed I would also like to see this page migrated away from inline forms. The problem is that it would be ideal to migrate all forms at the same time, and not have them be a mix of Field and Inline. What do you think it would be best here? To move forward with this change and then migrate to Field or do the whole thing?

**Comment by tskarhed** [comment]
@matyax I'm fine with merging this change the way it is now, as you seem keen on moving away from inline forms soon. I am not sure how much work it would require either, so it might be more efficient to do it in an internal PR. This PR is a quick win for now :)

**Comment by matyax** [comment]
Agreed. Also this is a sensitive area, so an internal PR would be my prefference.

**Comment by shabeebk** [comment]
So, out of my curiosity, is this going to be merged? 🤔 😄

**Comment by matyax** [comment]
We will review it as per usual: as soon as we can. We'll keep you updated, thanks for your patience 🙂

File: public/app/plugins/datasource/loki/configuration/QuerySettings.tsx [hunk 1]
```diff
@@ -27,64 +25,64 @@ export const QuerySettings = (props: Props) => {
...
+            limit to have a bigger result set for ad-hoc analysis. Decrease this limit if your browser becomes sluggish
+            when displaying the log results.
+          </>
+        }
+      >
+        <Input
+          type="number"
+          id="loki_config_maxLines"
+          value={maxLines}
+          onChange={(event: { currentTarget: { value: string } }) => onMaxLinedChange(event.currentTarget.value)}
```
Comment (matyax) [inline public/app/plugins/datasource/loki/configuration/QuerySettings.tsx:44]: Please use a proper type for the Event, there's no need to define a custom one.

File: public/app/plugins/datasource/loki/configuration/QuerySettings.tsx [hunk 2]
```diff
@@ -27,64 +25,64 @@ export const QuerySettings = (props: Props) => {
...
-                  {
-                    'Predefined operations are used as an initial state for your queries. They are useful, if you want to unpack, parse or format all log lines. Currently we support only log operations starting with |. For example: | unpack | line_format "{{.message}}".'
-                  }
-                </div>
+          >
+            <Input
+              type="string"
+              id="loki_config_predefinedOperations"
+              value={predefinedOperations}
+              onChange={(event: { currentTarget: { value: string } }) =>
```
Comment (matyax) [inline public/app/plugins/datasource/loki/configuration/QuerySettings.tsx:69]: Same here.

**Review by matyax — CHANGES_REQUESTED** [review]
Works and looks good! Please fix the event types and it's ready.

**Comment by shabeebk** [comment]
@matyax Thanks for the review and suggestion. I don't know if I got your comment correct. Anyway updated the type in the form event and pushed a new commit.

File: public/app/plugins/datasource/loki/configuration/QuerySettings.tsx (hunk 1 above)
Comment (shabeebk) [reply public/app/plugins/datasource/loki/configuration/QuerySettings.tsx]: Changed

File: public/app/plugins/datasource/loki/configuration/QuerySettings.tsx (hunk 2 above)
Comment (shabeebk) [reply public/app/plugins/datasource/loki/configuration/QuerySettings.tsx]: Changed

**Review by matyax — APPROVED** [review]

**Comment by matyax** [comment]
Thank you @shabeebk , great work!

**Comment by shabeebk** [comment]
Thanks.

**Pull request merged**